### PR TITLE
Remove `m_*` functions

### PR DESCRIPTION
**Pull request by j-c-c**

### CODE REVIEW

File: src/aspire/basis/ffb_3d.py [hunk 1]
```diff
@@ -60,15 +59,15 @@ def _precomp(self):
...
         z, wt_z = lgwt(n_phi, -1, 1, dtype=self.dtype)
-        r = m_reshape(xp.asarray(r), (n_r, 1))
+        r = xp.asarray(r.reshape(n_r, 1))
         rh = xp.asnumpy(r)
-        wt_r = m_reshape(xp.asarray(wt_r), (n_r, 1))
-        z = m_reshape(xp.asarray(z), (n_phi, 1))
-        wt_z = m_reshape(xp.asarray(wt_z), (n_phi, 1))
+        wt_r = xp.asarray(wt_r.reshape(n_r, 1))
+        z = xp.asarray(z.reshape(n_phi, 1))
+        wt_z = xp.asarray(wt_z.reshape(n_phi, 1))
```
Comment (j-c-c) [inline src/aspire/basis/ffb_3d.py]: reshape on gpu

File: src/aspire/basis/fb_2d.py [hunk 2]
```diff
@@ -220,32 +218,28 @@ def _evaluate(self, v):
...
             of `basis`. This is an array of vectors whose last dimension equals
             `self.count` and whose first dimensions correspond to
             first dimensions of `v`.
         """
-        v = v.T
-        x, sz_roll = unroll_dim(v, self.ndim + 1)
-        x = m_reshape(
-            x, new_shape=tuple([np.prod(self.sz)] + list(x.shape[self.ndim :]))
-        )
+        x = x.reshape(x.shape[0], -1)
```
Comment (j-c-c) [inline src/aspire/basis/fb_2d.py:232]: Should we be keeping track of stack shape here?

Comment (j-c-c) [reply src/aspire/basis/fb_2d.py:232]: No, handled by `Basis` base class.

File: src/aspire/basis/fb_3d.py [hunk 3]
```diff
@@ -150,21 +148,17 @@ def _evaluate(self, v):
             This is an array whose first dimensions equal `self.z` and the
             remaining dimensions correspond to dimensions two and higher of `v`.
         """
-
-        v = v.T
-        v, sz_roll = unroll_dim(v, 2)
-
+        stack_shape = v.shape[:-1]
```
Comment (j-c-c) [inline src/aspire/basis/fb_3d.py:151]: Don't need this stack_shape. Handled by base class

Comment (j-c-c) [reply src/aspire/basis/fb_3d.py:151]: done

File: src/aspire/basis/ffb_3d.py (hunk 1 above)
Comment (j-c-c) [reply src/aspire/basis/ffb_3d.py]: done

File: src/aspire/reconstruction/mean.py [hunk 4]
```diff
@@ -139,7 +139,7 @@ def _compute_kernel(self):
                     if j != k:
                         kernel[j, k] += batch_kernel
 
-        kermat_f = np.zeros((self.r, self.r, _2L, _2L, _2L))
+        kermat_f = np.zeros((self.r, self.r, _2L, _2L, _2L), dtype=self.dtype)
```
Comment (j-c-c) [inline src/aspire/reconstruction/mean.py]: Note: Found this while switching out `FBBasis3D` for `FFBBasis3D` in the Cov3D tutorial.

File: src/aspire/basis/ffb_3d.py [hunk 5]
```diff
@@ -244,37 +240,35 @@ def _evaluate(self, v):
...
 
-        u_even = m_reshape(u_even, (2 * self.ell_max + 1, n_phi * n_r * n_data))
-        u_odd = m_reshape(u_odd, (2 * self.ell_max + 1, n_phi * n_r * n_data))
+        u_even = u_even.reshape(n_data * n_r * n_phi, 2 * self.ell_max + 1)
+        u_odd = u_odd.reshape(n_data * n_r * n_phi, 2 * self.ell_max + 1)
 
         # evaluate the theta parts
-        w_even = self._precomp["ang_theta_wtd"] @ u_even
-        w_odd = self._precomp["ang_theta_wtd"] @ u_odd
+        w_even = u_even @ self._precomp["ang_theta_wtd"].T
```
Comment (garrettwrong) [inline src/aspire/basis/ffb_3d.py:264]: nice, glad this line worked out to cancel with the pf tranpose that was below.

File: src/aspire/reconstruction/mean.py (hunk 4 above)
Comment (garrettwrong) [reply src/aspire/reconstruction/mean.py]: I think this will be covered before this PR will go in.

File: src/aspire/basis/fb_2d.py [hunk 6]
```diff
@@ -189,7 +187,7 @@ def _evaluate(self, v):
```
Comment (garrettwrong) [inline src/aspire/basis/fb_2d.py:187]: same question as evaluate_t, do we still need to reshape `v` here?

File: src/aspire/basis/fb_2d.py [hunk 7]
```diff
@@ -220,32 +218,28 @@ def _evaluate(self, v):
```
Comment (garrettwrong) [inline src/aspire/basis/fb_2d.py:218]: why not build x as (v.shape[0], *self.sz), is it slower?

File: src/aspire/basis/fb_2d.py (hunk 2 above)
Comment (garrettwrong) [inline src/aspire/basis/fb_2d.py:232]: Why is this required.  `evaluate_t` sends something shaped like `(n, count)` correct?

File: src/aspire/basis/fb_3d.py (hunk 3 above)
Comment (garrettwrong) [inline src/aspire/basis/fb_3d.py:151]: FWIW, I think the intent of `Basis.evaluate{_t}` was to handle this business so that `_evaluate{_t}` would always get a single stack axis.  

I see where you were ran up against the `mat_evaluate{_t}` in cov3d calling the private method.  Probably could have converted that towards the public method, but its okay to punt on this. I'll get it when I finish the `m_` work.

**Review by garrettwrong — CHANGES_REQUESTED** [review]
Thanks for moving this forward.  Had a couple questions (really just two, but one shows up in a few places).

Almost done with the `m_*` ! 😀

File: src/aspire/basis/fb_3d.py (hunk 3 above)
Comment (j-c-c) [reply src/aspire/basis/fb_3d.py:151]: Ok, yeah, I tried switching to the public method (which worked fine) but figured I'd wait for feedback on it. Thanks.

File: src/aspire/basis/fb_2d.py (hunk 6 above)
Comment (j-c-c) [reply src/aspire/basis/fb_2d.py:187]: no, we do not. removed. thanks

File: src/aspire/basis/fb_2d.py (hunk 7 above)
Comment (j-c-c) [reply src/aspire/basis/fb_2d.py:218]: yes, it's significantly slower.

File: src/aspire/basis/fb_2d.py (hunk 2 above)
Comment (j-c-c) [reply src/aspire/basis/fb_2d.py:232]: `evaluate_t` is sending something shaped `(n, *self.sz)`, ie. a stack-flattened set of images, so this reshape is actually doing something.

File: src/aspire/basis/ffb_3d.py (hunk 5 above)
Comment (j-c-c) [reply src/aspire/basis/ffb_3d.py:264]: 😇

File: src/aspire/basis/fb_3d.py (hunk 3 above)
Comment (garrettwrong) [reply src/aspire/basis/fb_3d.py:151]: okay, np, i'll get it with the 3d `m_*` changes.

File: src/aspire/basis/fb_2d.py (hunk 7 above)
Comment (garrettwrong) [reply src/aspire/basis/fb_2d.py:218]: Oh okay, bummer, thanks!

**Review by garrettwrong — APPROVED** [review]
Great, thanks!

File: src/aspire/basis/fb_2d.py (hunk 2 above)
Comment (janden) [inline src/aspire/basis/fb_2d.py:232]: Looks like the stack shape is lost here (it's not saved or restored later in `v`).

File: src/aspire/basis/fb_3d.py [hunk 8]
```diff
@@ -176,43 +170,36 @@ def _evaluate(self, v):
...
             equals `self.count` and whose remaining dimensions correspond
             to higher dimensions of `v`.
         """
-        v = v.T
-        x, sz_roll = unroll_dim(v, self.ndim + 1)
-        x = m_reshape(
-            x, new_shape=tuple([np.prod(self.sz)] + list(x.shape[self.ndim :]))
-        )
-
+        stack_shape = x.shape[: -self.ndim]
```
Comment (janden) [inline src/aspire/basis/fb_3d.py:192]: We keep repeating this unroll/roll pattern, which makes me wonder why we needed to remove it in the first place…?

**Review by janden — CHANGES_REQUESTED** [review]
Looks great! Just two things.

File: src/aspire/basis/fb_2d.py (hunk 2 above)
Comment (j-c-c) [reply src/aspire/basis/fb_2d.py:232]: Yeah, I had the same thought, but the stack is handled by the public method `evaluate_t`. It's actually passed to this private method already stack-flattened.

File: src/aspire/basis/fb_3d.py (hunk 8 above)
Comment (j-c-c) [reply src/aspire/basis/fb_3d.py:192]: If you're referring to the `unroll_dim/roll_dim` methods, I removed them because they are using `m_reshape` under the hood.

File: src/aspire/basis/fb_2d.py (hunk 2 above)
Comment (janden) [reply src/aspire/basis/fb_2d.py:232]: Ok got it.

**Review by janden — APPROVED** [review]
